File: Examples/SharedFiles/Models/Step.m
//
//  Step.m
//  NordicCook
//
//  Created by Nahuel Marisi on 17/02/2015.
//  Copyright (c) 2015 TechBrewers. All rights reserved.
//

#import "Step.h"
#import "Macros.h"
#import "IngredientsSorter.h"

@implementation Step

- (NSArray *)sortedIngredients
{
  return [IngredientsSorter sortedIngredientsArrayFromArray:self.ingredients];
}

- (NSString *)description
{
  return description(SelfKeyPath(number),
                     SelfKeyPath(text),
                     SelfKeyPath(ingredients),
										 SelfKeyPath(imagePath));
}

@end

File: Examples/SwiftExample/SwiftExample/Models/Step.swift
//
//  Step.swift
//  SwiftExample
//
//  Created by Nahuel Marisi on 2015-09-03.
//  Copyright © 2015 TechBrewers. All rights reserved.
//

import Foundation

@objc class Step: NSObject {
    
    var text: String
    var ingredients:[Ingredient]
    
    init(text: String, ingredients: [Ingredient]) {
        self.text = text
        self.ingredients = ingredients
    }
    
    override init() {
        self.text = ""
        self.ingredients = Array<Ingredient>()
        super.init()
    }

}